CONTRIBUTING.md

```diff
@@ -0,0 +1,466 @@
+  "not-ready").
+
+- Always add value. Don't restate what is already indicated by the type signature.
+
+- Describe units of measure and integrity constraints of variables.
+
+- Link to conceptual documentation for more complete descriptions of how APIs fit together as a
+  whole.
+
+---
+
+## Documentation Style
+
+It is important to create documentation that follows similar guidelines. This allows documentation
+to be clear and concise while allowing users to easily find necessary information.
+
+These are general style guidelines that can help create clearer documentation:
+
+- **Write in plain U.S. English.** You should write in plain U.S. English and try to avoid over-complicated words when you describe something.
+
+- **Avoid using pronouns such as "I" or "we".** These can be quite ambiguous when someone reads the
+  documentation. It is better to say "You should do…." instead of "We recommend that you do….". It
+  is OK to use "you" as this allows the documentation to speak to a user.
+
+- **If you plan on using acronyms, you should define them the first time you write about them.** For
+  example, looks good to me (LGTM). Don't assume that everyone will understand all acronyms. You do
+  not need to define acronyms that might be considered industry standards such as TCP/IP.
+
+- **In most cases, avoid future tense.** Words such as "will" are very ambiguous. For example "you
+  will see" can lead to questions such as "when will I see this?". In 1 minute or in 20 minutes? In
+  most cases, assume that when someone reads the documentation you are sitting next to them and
+  reading the instructions to them.
+
+- **Use active voice.** You should always try to write in the active voice since passive voice can
+  make sentences very ambiguous and hard to understand. There are very few cases where you should
+  use the passive voice for technical documentation.
+
+  - Active voice - the subject performs the action denoted by the verb.
+
+    - "The operating system runs a process." This sentence answers the question on what is
+      happening and who/what is performing the action.
+
+  - Passive voice - the subject is no longer _active_, but is, instead, being acted upon by the
+    verb - or passive.
+
+    - "A process is being run." This sentence is unclear about who or what is running the process.
+      You might consider "a process is run by the operating system", but the object of the action
+      is still made into the subject of the sentence, which indicates passive voice. Passive voice
+      tends to be wordier than active voice, which can make your sentence unclear. In most cases,
+      if you use "by" this indicates that your sentence might be still be in passive voice. A
+      correct way of writing this example is, "The operating systems runs the process."
+
+- **Do not list future plans for a product/feature.** "In the future, the product will have no
```

Comment:
This will seem hard to adhere to for us.

Reply:
This seems important for us to enforce within our world of documentation. I think we have been very bad in the past talking about features like off-chain workers, which is _still_ not ready.

I am a big fan of this rule. Anything speculative can live with the code or in issues.

Reply:
Who says when a feature is ready? Where do we draw the line on what is speculative vs non-speculative? I'm just saying that these things aren't so clear. We're working with speculative technology and documenting speculative code.

Also, I think raising awareness of offchain workers before was useful even if it isn't completely finished. Because of this conceptual documentation, some developers are positioning themselves to take advantage of this feature when it is ready and that's what we want.

Reply:
Isn't versioning supposed to solve part of this issue? A feature could be tested on a specific Substrate version.
For the second part, I'll also prefer to be less explicit about future features until we explore and play with them.

Reply:
So, all docs are limited to code that is feature complete now? We're going to keep getting a better understanding of what a good runtime architecture will look like and I don't feel comfortable keeping this to ourselves until we get around to writing usage-oriented tutorials. 

For example, fixed point arithmetic is still undergoing refactoring. The storage api for child tries isn't completely finished nor are all parts of off-chain workers. Swappable consensus will not be ready for some time. Also, we keep claiming that substrate chains can trivially be turned into parachains but cumulus is also still a wip.

So, yes, maybe we are less loud about these features but we should continue communicating our expectations of future best practices in conceptual documentation, even if we don't have immediate procedural examples. *I'm mostly just being pedantic because I don't think this can be enforced consistently, but it's fine to discuss later on a case by case basis.*